Refractor resultwatchers

clample opened the pull request.

Commit:
Refractor result watchers

- Rename Util class to a more specific name. Make the class no longer public since it will only be used in our TestWatcher.
- Seperate intermediate and suite watchers
- Delete outdated test. Test was based on old version of suites api.
- Add intermediate JUnit test
- Add isTestLocal parameter to all watchers. This makes sure results are reported when running against On-Premise or Local instance of TestObject.

Commit:
Remove TestObject capabilities

We already have a TestObjectCapabilities class in testobject-java-api. I've added all of the constants to the other class and deleted this one.

Commit:
Bump version number and update dependency

We require changes to testobject-java-api in version 0.0.26

clample commented:
travis failed because this version uses testobject-java-api version 0.0.26 (not yet released) which includes the changes here: https://github.com/testobject/testobject-java-api/pull/17

crummy commented on src/main/java/org/testobject/appium/junit/TestObjectAppiumSuite.java:
```diff
@@ -127,7 +128,8 @@ public TestObjectAppiumSuite(Class<?> clazz) throws InitializationError {
 
 			Optional<String> endpointFromEnvironment = Env.getApiEndpoint();
-			String testObjectApiEndpoint = endpointFromEnvironment.isPresent() ? endpointFromEnvironment.get() : config.testObjectApiEndpoint();
+			String endpointFromConfig = config.testObjectApiEndpoint().equals("") ? TESTOBJECT_API_ENDPOINT.toString() : config.testObjectApiEndpoint();
```
Instead of `.equals("")` you can check `.isEmpty()`

crummy commented on src/main/java/org/testobject/appium/junit/TestObjectAppiumSuiteWatcher.java:
```diff
@@ -0,0 +1,108 @@
+        this.test = TestParser.from(description);
+    }
+
+    @Override protected void succeeded(Description description) {
```
Usually we put @Override on a separate line to the method name. Applies several times in this class

crummy commented on src/main/java/org/testobject/appium/junit/TestObjectAppiumSuiteWatcher.java:
```diff
@@ -0,0 +1,108 @@
+
+        Optional<TestReport.Id> testReportId = reporter.suiteReport().getTestReportId(this.test);
+
+        if (testReportId.orNull() == null) {
```
I think you can check for isPresent() here instead

clample replied:
Sadly I don't think that this optional library has that

crummy commented on src/main/java/org/testobject/appium/junit/TestObjectAppiumSuiteWatcher.java:
```diff
@@ -0,0 +1,108 @@
+            throw new IllegalStateException("test report not present");
+        }
+
+        return testReportId.orNull().toString();
```
Might as well use .get() instead of .orNull(); you already null checked earlier

crummy commented on src/main/java/org/testobject/appium/SuiteReporter.java:
```diff
@@ -0,0 +1,47 @@
+
+    private void updateSuiteReport(SuiteReport suiteReport, Test test, boolean passed) {
+        Optional<TestReport.Id> testReportId = suiteReport.getTestReportId(test);
+        if (testReportId.orNull() == null) {
```
You can use .isPresent() instead I think

crummy commented on src/main/java/org/testobject/appium/junit/TestObjectTestResultWatcher.java:
```diff
@@ -1,186 +1,61 @@
 
-	public TestObjectTestResultWatcher(String baseUrl) {
-		this.baseUrl = baseUrl;
+	public TestObjectTestResultWatcher(boolean isTestLocal) {
```
Do folks no longer have the ability to set a custom URL? What if their local Appium server is on a non-standard port?

You could consider having an overloaded constructor that takes a URL or string, or maybe a factory with methods like .create(), .createLocal(), create(String)

edit: nevermind the other constructors are below

crummy commented on src/test/java/org/testobject/appium/junit/AppiumDriverCalculatorIntermediateTestJUnit.java:
```diff
@@ -0,0 +1,52 @@
+    public void setup() {
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+
+        capabilities.setCapability(TestObjectCapabilities.TESTOBJECT_API_KEY, "7CDE94EFFE3E4EF4A773DB2728688C53");
```
Can folks see this if they decompile the jar? Do we care?

Commit:
Reformat

- use .isEmpty() rather than .equals("") for readability
- reformat the @Oberride annotations
